spreadsheet_oca/static/src/spreadsheet/bundle/spreadsheet_action.esm.js

```diff
@@ -176,6 +176,8 @@ export class ActionSpreadsheetOca extends Component {
             metaData: {
                 colGroupBys: this.import_data.metaData.colGroupBys,
                 rowGroupBys: this.import_data.metaData.rowGroupBys,
+                expandedColGroupBys: this.import_data.metaData.expandedColGroupBys,
```

Comment:
After testing, I had to set colGroupBys and rowGroupBys as below instead, otherwise the pivot is not displayed correctly as a full range. It works for me without the need for seperately setting expandedColGroupBys / expandedRowGroupBys as parameters:

```
colGroupBys: this.import_data.metaData.expandedColGroupBys,
rowGroupBys: this.import_data.metaData.expandedRowGroupBys,
```

Reply:
With this, the initial state is lost, try to set the pivot without doing any change to pivot view and you will see what i mean

Reply:
Then I think you should check if `expandedColGroupBys` is set (or has value), and if not, set `his.import_data.metaData.colGroupBys` instead.

Reply:
Yes I had missed that the initial state was lost, thanks @CarlosRoca13 and @pedrobaeza 

Tried this and seems to work ok:

```
        const colGroupBys = this.import_data.metaData.expandedColGroupBys.length > 0 ? this.import_data.metaData.expandedColGroupBys : this.import_data.metaData.colGroupBys;
        const rowGroupBys = this.import_data.metaData.expandedRowGroupBys.length > 0 ? this.import_data.metaData.expandedRowGroupBys : this.import_data.metaData.rowGroupBys;
        const pivot_info = {
            metaData: {
                colGroupBys: colGroupBys,
                rowGroupBys: rowGroupBys,
                activeMeasures: this.import_data.metaData.activeMeasures,
                resModel: this.import_data.metaData.resModel,
                fields: this.import_data.metaData.fields,
            },
            searchParams: this.import_data.searchParams,
        };
```

Reply:
It seems to be working ok with that code, but it isn't. If we expand the default pivot state, we will be loosing the initial state too.

Example:
![No_OK](https://github.com/OCA/spreadsheet/assets/35952655/1b749083-2cc8-447b-8931-f0a8a8480d43)
You can see that the dates are disappearing 

I do not understand the need to remove those 2 keys when with them the pivot view is working properly....

Example:
![OK](https://github.com/OCA/spreadsheet/assets/35952655/230dafba-264c-40b9-b7f5-43021345311e)


Right now the only problem is to set the data of comparison, because the data is computed and IDK how to reach that info